src/Adapter/Cart/CommandHandler/AddProductToCartHandler.php

```diff
@@ -86,7 +88,9 @@ public function handle(AddProductToCartCommand $command): void
         $cart = $this->getCart($command->getCartId());
         $product = $cart->getProductQuantity($productIdValue, $combinationId, $customizationId);
 
-        $quantity = $command->getQuantity() + (int) $product['quantity'];
+        $cartQuantity = (int) $product['quantity'] - $this->getProductGiftedQuantity($cart, $productIdValue, $combinationId);
```

Comment:
Oh for adding a product But what happens if we update the quantity directly (by clicking on the arrow in the cart product list) Isn't there a weird offset due to this gift product as well?

In which case maybe this substraction should be handled in `UpdateProductQuantityInCartHandler` so that it covers more use cases

I'm surprised this was not handled in the legacy class already, does it mean this can happen on the FO as well?

Reply:
@jolelievre you mean decrease or increase quantity of gift product?

Reply:
Yes decrease/increase in the list, or even simply update

Reply:
you cannot increase/decrease quantity of the gift itself (you don't have the quantity input), you can only increase/decrease other products.
The bug I detected in the previous comment https://github.com/PrestaShop/PrestaShop/pull/21781#issuecomment-722551891 is that the other products increase/decrease doesn't respect the minimum quantity : No exception thrown